iib/web/api_v1.py

```diff
@@ -438,25 +443,46 @@ def regenerate_bundle_batch():
     request_jsons = []
     # This list will be used for the log message below and avoids the need of having to iterate
     # through the list of requests another time
-    request_id_strs = []
-    for build_request, request in zip(payload['build_requests'], requests):
-        request_jsons.append(request.to_json())
-        request_id_strs.append(str(request.id))
-
-        error_callback = failed_request_callback.s(request.id)
-        handle_regenerate_bundle_request.apply_async(
-            args=[
-                build_request['from_bundle_image'],
-                build_request.get('organization'),
-                request.id,
-            ],
-            link_error=error_callback,
-            queue=_get_user_queue(),
+    processed_request_ids = []
+    build_and_requests = zip(payload['build_requests'], requests)
+    try:
+        for build_request, request in build_and_requests:
+            error_callback = failed_request_callback.s(request.id)
+            handle_regenerate_bundle_request.apply_async(
+                args=[
+                    build_request['from_bundle_image'],
+                    build_request.get('organization'),
+                    request.id,
+                ],
+                link_error=error_callback,
+                queue=_get_user_queue(),
+            )
+
+            request_jsons.append(request.to_json())
+            processed_request_ids.append(str(request.id))
+    except OperationalError:
+        failed_ids = []
+        for request in requests:
+            if str(request.id) in processed_request_ids:
+                continue
+
+            failed_ids.append(str(request.id))
+            request.add_state('failed', 'The scheduling of the build request failed.')
```

Comment:
The rest of the state reasons in the code do not have periods at the end, so let's remove that to be consistent.
```suggestion
            request.add_state('failed', 'The scheduling of the build request failed')
```